frontend/src/components/common/Resource/ResourceListView.tsx

```diff
@@ -1,49 +1,36 @@
 import React, { PropsWithChildren } from 'react';
-import { useTranslation } from 'react-i18next';
-import { ActionButton } from '..';
+import { KubeObject } from '../../../lib/k8s/cluster';
 import SectionBox from '../SectionBox';
 import SectionFilterHeader, { SectionFilterHeaderProps } from '../SectionFilterHeader';
-import ResourceTable, {
-  ResourceTableFromResourceClassProps,
-  ResourceTableProps,
-} from './ResourceTable';
+import ResourceTable, { ResourceTableProps } from './ResourceTable';
 
-export interface ResourceListViewProps extends PropsWithChildren<ResourceTableProps> {
+export interface ResourceListViewProps<ItemType>
+  extends PropsWithChildren<ResourceTableProps<ItemType>> {
   title: string | JSX.Element;
   headerProps?: Omit<SectionFilterHeaderProps, 'title'>;
 }
 
-export interface ResourceListViewWithResourceClassProps
-  extends Omit<ResourceListViewProps, 'data'> {
-  resourceClass: ResourceTableFromResourceClassProps['resourceClass'];
+type Class<T> = new (...args: any[]) => T;
+
+export interface ResourceListViewWithResourceClassProps<ItemType>
+  extends Omit<ResourceListViewProps<ItemType>, 'data'> {
+  resourceClass: Class<ItemType>;
 }
 
-export default function ResourceListView(
-  props: ResourceListViewProps | ResourceListViewWithResourceClassProps
+export default function ResourceListView<ItemType>(
+  props: ResourceListViewProps<ItemType> | ResourceListViewWithResourceClassProps<ItemType>
 ) {
   const { title, children, headerProps, ...tableProps } = props;
-  const { t } = useTranslation();
-  const [columnChooserAnchorEl, setColumnChooserAnchorEl] = React.useState<null | HTMLElement>(
-    null
-  );
-  const withNamespaceFilter = (props as ResourceListViewWithResourceClassProps).resourceClass
-    ?.isNamespaced;
+  const withNamespaceFilter = (
+    (props as ResourceListViewWithResourceClassProps<KubeObject>).resourceClass as KubeObject
+  )?.isNamespaced;
```

Comment:
This was casted already, even if we got rid of any casting here it still wouldn't really change much since KubeObject resolves to `any`. It can be improved though, I want to take a look at that after finishing with the table